smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/IncomingKafkaRecord.java

```diff
@@ -81,4 +96,8 @@ public Metadata getMetadata() {
     public CompletionStage<Void> nack(Throwable reason) {
         return onNack.handle(this, reason);
     }
+
+    public void injectTracingMetadata(TracingMetadata tracingMetadata) {
```

Comment:
with metadata not final anymore, we must synchronize access.

Reply:
do you mean just making the method synchronized, or something different?

Reply:
yes, `synchronized`